authorize.go

```diff
@@ -77,8 +74,14 @@ func (a *autographer) authorize(r *http.Request, body []byte) (userid string, au
 		}
 	}
 	if err != nil {
-		return "", false, err
+		return nil, "", false, err
 	}
+	return auth, userid, true, nil
+}
+
+// authorizeBody validates the body within the request and returns
+// a boolean indiciating whether its valid.
+func (a *autographer) authorizeBody(auth *hawk.Auth, r *http.Request, body []byte) (authorize bool, err error) {
```

Comment:
same things here docstring out of date and can combine authorize and err return args.